packages/@react-native-windows/cli/src/config/dependencyConfig.ts

```diff
@@ -271,47 +271,69 @@ export function dependencyConfigWindows(
+      if (projectType === 'dynamiclibrary' || projectType === 'winmdobj') {
+        const projectLang = configUtils.getProjectLanguage(projectFile);
 
-      const projectNamespace = configUtils.getProjectNamespace(projectContents);
+        const projectName = configUtils.getProjectName(
+          projectFile,
+          projectContents,
+        );
 
-      const directDependency = true;
+        const projectGuid = configUtils.getProjectGuid(projectContents);
 
-      const cppHeaders: string[] = [];
-      const cppPackageProviders: string[] = [];
-      const csNamespaces: string[] = [];
-      const csPackageProviders: string[] = [];
+        const projectNamespace = configUtils.getProjectNamespace(
+          projectContents,
+        );
 
-      if (projectNamespace !== null) {
-        const cppNamespace = projectNamespace.replace(/\./g, '::');
-        const csNamespace = projectNamespace.replace(/::/g, '.');
+        const directDependency = true;
 
-        cppHeaders.push(`winrt/${csNamespace}.h`);
-        cppPackageProviders.push(`${cppNamespace}::ReactPackageProvider`);
-        csNamespaces.push(`${csNamespace}`);
-        csPackageProviders.push(`${csNamespace}.ReactPackageProvider`);
-      }
+        const cppHeaders: string[] = [];
+        const cppPackageProviders: string[] = [];
+        const csNamespaces: string[] = [];
+        const csPackageProviders: string[] = [];
 
-      result.projects.push({
-        projectFile: path.relative(sourceDir, projectFile),
-        projectName,
-        projectLang,
-        projectGuid,
-        directDependency,
-        cppHeaders,
-        cppPackageProviders,
-        csNamespaces,
-        csPackageProviders,
-      });
+        if (projectNamespace !== null) {
+          const cppNamespace = projectNamespace.replace(/\./g, '::');
+          const csNamespace = projectNamespace.replace(/::/g, '.');
+
+          cppHeaders.push(`winrt/${csNamespace}.h`);
+          cppPackageProviders.push(`${cppNamespace}::ReactPackageProvider`);
+          csNamespaces.push(`${csNamespace}`);
+          csPackageProviders.push(`${csNamespace}.ReactPackageProvider`);
+        }
+
+        result.projects.push({
+          projectFile: path.relative(sourceDir, projectFile),
+          projectName,
+          projectLang,
+          projectGuid,
+          directDependency,
+          cppHeaders,
+          cppPackageProviders,
+          csNamespaces,
+          csPackageProviders,
+        });
+      } else {
+        const projectPath = path.relative(sourceDir, projectFile);
+        result.projects.push({
+          projectFile: `Error: ${projectPath} is type '${projectType}'`,
+          directDependency: false,
+          projectName: '',
+          projectLang: null,
+          projectGuid: null,
+          cppHeaders: [],
+          cppPackageProviders: [],
+          csNamespaces: [],
+          csPackageProviders: [],
+        });
+      }
```

Comment:
I'm not sure pushing a project onto the config if it isn't of interest is the right thing? @jonthysell what do you think?

Also note that the project type will be empty for some projects, e.g. WAP projects

Reply:
Pushing the error invalidates the project here: https://github.com/microsoft/react-native-windows/blob/b7da9b7b910ee38853a4afdcdbad4afde9ab6958/packages/%40react-native-windows/cli/src/runWindows/utils/autolink.ts#L443-L450

It's a pattern that is used elsewhere: https://github.com/microsoft/react-native-windows/blob/b7da9b7b910ee38853a4afdcdbad4afde9ab6958/packages/%40react-native-windows/cli/src/config/projectConfig.ts#L175-L185

If you feel uncomfortable with this change, an alternative would be to downgrade this to a warning instead of throwing an exception: https://github.com/microsoft/react-native-windows/blob/b7da9b7b910ee38853a4afdcdbad4afde9ab6958/packages/%40react-native-windows/cli/src/runWindows/utils/autolink.ts#L428-L437